flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java

```diff
@@ -226,6 +227,13 @@ public void createTable(ObjectPath tablePath, UpdateSchema updateSchema, boolean
                     e);
         }
         Table table = newHmsTable(tablePath);
+
+        if (hiveConf.getEnum(TABLE_TYPE.key(), TableType.MANAGED_TABLE)
```

Comment:
It's better to move this logic to `newHmsTable` which initialzes the `Table` and sets the parameters of the `Table`.

Reply:
done